wandb/util.py

```diff
@@ -1165,6 +1166,16 @@ def uri_from_path(path):
     return url.path if url.path[0] != "/" else url.path[1:]
 
 
+def _has_internet():
+    """Attempts to open a DNS connection to Googles root servers"""
+    try:
+        s = socket.create_connection(("8.8.8.8", 53), 0.5)
+        s.close()
+        return True
+    except OSError:
+        return False
```

Comment:
@raubitsj this is HOT 🔥 .  We can likely get rid of the hacky thread timeout nonsense in the user process...